api/handlers_spaces.go

```diff
@@ -40,12 +45,36 @@ func (s *server) SpaceGetHandler(w http.ResponseWriter, r *http.Request) {
 	spaceID := vars["space"]
 	log.Debugf("getting costs for space %s", spaceID)
 
-	y, m, d := time.Now().Date()
+	// Did we get cost-explorer start and end times on the API?
+	// todo:
+	//   verify that end date is after start date.
+	//   write unit test
+	if endTime == "" || startTime == "" {
+		// if it's the first day of the month, get today's usage thus far
+		y, m, d := time.Now().Date()
+		if d == 1 {
+			d = 2
+		}
+
+		endTime = fmt.Sprintf("%d-%02d-%02d", y, m, d)
+		startTime = fmt.Sprintf("%d-%02d-01", y, m)
+
+	} else {
+		sTime, err := time.Parse("1066-01-01", startTime)
```

Comment:
Is 1066 the standard format for a year or it doesn't matter?

Reply:
ancient history - year of norman invasion.   I replaced that parse with the "special layout parameter":
https://golang.org/pkg/time/#Parse
https://yourbasic.org/golang/format-parse-string-time-date-example/
"(This date is easier to remember when written as 01/02 03:04:05PM ‘06 -0700.)"